src/platform/PICO/dshot_pico.c

```diff
@@ -0,0 +1,412 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <math.h>
+#include <string.h>
+
+#include "platform.h"
+
+#ifdef USE_DSHOT
+
+#include "build/debug.h"
+#include "build/debug_pin.h"
+
+#include "drivers/io.h"
+#include "drivers/dshot.h"
+#include "drivers/dshot_command.h"
+#include "drivers/dshot_command.h"
+#include "drivers/motor_types.h"
+
+#include "drivers/time.h"
+#include "drivers/timer.h"
+
+#include "pg/motor.h"
+
+#include "hardware/pio.h"
+#include "hardware/clocks.h"
+
+// Maximum time to wait for telemetry reception to complete
+#define DSHOT_TELEMETRY_TIMEOUT 2000
+
+FAST_DATA_ZERO_INIT timeUs_t dshotFrameUs;
+//static FAST_DATA_ZERO_INIT timeUs_t lastSendUs;
+
+typedef struct motorOutput_s {
+    int offset;
+    dshotProtocolControl_t protocolControl;
+    int pinIndex;            // pinIndex of this motor output within a group that bbPort points to
+    IO_t io;                 // IO_t for this output
+    bool configured;
+    bool enabled;
+    PIO pio;
+    uint16_t pio_sm;
+} motorOutput_t;
+
+typedef struct picoDshotTelemetryBuffer_s {
+    union {
+        uint32_t u32[2];
+        uint16_t u16[4];
+    };
+} picoDshotTelemetryBuffer_t;
+
+static motorProtocolTypes_e motorProtocol;
+//uint8_t dshotMotorCount = 0;
```

Comment:
Can we use motorCount here, or do we require separate pwmMotorCount and dshotMotorCount variables?